Added thermal image converter (doesn't work)

Dudi512 opened the pull request.

Dudi512 committed:
Added thermal image converter (doesn't work)

MaxKolac commented:
Bruh, jeśli nie działa to nie merguj jeszcze.

MaxKolac commented on client/src/FormMain.cs:
```diff
@@ -491,6 +511,41 @@ private void updateStateList()
+            return image;
+        }
+
+        private string tryGetSolutionDirectoryInfo(string currentPath = null)
+        {
+            var directory = new DirectoryInfo(
+                currentPath ?? Directory.GetCurrentDirectory());
+            while (directory != null && !directory.GetFiles("*.sln").Any())
+            {
+                directory = directory.Parent;
+            }
+            return directory.ToString();
+        }
```
Zamiast całej tej metody, spróbuj użyć `Path.Combine(Environment.CurrentDirectory)`.

Dudi512 replied:
Samo Path.Combine(Environment.CurrentDirectory) zwraca następującą ścieżkę:
"D:\ZPI\client\src\bin\Debug\net7.0-windows10.0.17763.0"

Zmodyfikowałem funkcję w następujący sposób:

        private string tryGetDirectory()
        {
            string currentDirectory = Environment.CurrentDirectory;
            string communicationModelsPath = Path.Combine(currentDirectory,"..", "..", "..", "..", "..", "communicationModels");
            string fullPath = Path.GetFullPath(communicationModelsPath);

            return fullPath;
        }

Co zwraca ścieżkę "D:\ZPI\communicationModels\", tak samo jak wcześniejsza funkcja.

MaxKolac replied:
https://github.com/MaxKolac/zpi/pull/26/commits/1e5943da90badb1a03ee4e1f6e8086a1a614ecf4

MaxKolac commented on communicationModels/tempFile.png:
Dodawanie plików tymczasowych do kodu źródłowego to słaby pomysł, lepiej to usunąć.

Dudi512 replied:
Done

MaxKolac commented on server/src/database.sqlite3:
Co dokładnie dodałeś do bazy danych?

Dudi512 replied:
Zmieniłem rekordy (adresy ip, typy urządzeń oraz porty) aby client łączył się prawidłowo z bazą. Ilość rekordów nie uległa zmianie.

MaxKolac replied:
Nie trzeba edytować bazy danych jaką przy kompilacji serwer kopiuje. Powinieneś te zmiany wprowadzać w bazie danych, która po kompilacji będzie znajdować się w folderze `bin`. W ten sposób nie będziemy sobie nawzajem nadpisywać preferowanych wartości w rekordach.

Dudi512 committed:
Changed tryGetDirectory function

Dudi512 committed:
Removed tempFile (again)

MaxKolac committed:
Added SQLiteStudio autogenerated files

MaxKolac committed:
Server wont mistakenly update database

..when receiving incorrect data.

MaxKolac committed:
Fixed console logs lookin weird

MaxKolac committed:
Fixed an exception thrown when extracting real image

Dwie sprawy:
1. Okazuje się że klasa Image ma kompletnie w poważaniu metadane przy zapisywaniu. Metoda `Image.Save` zapisywała plik, który był faktycznie zdjęciem... ale bez metadanych. Jego rozmiar to był 28 KB, podczas gdy zdjęcia w folderze ExampleImages na ZPICameraSimulator mają po 2 MB. Dlatego Result tego jednego Task'a to był pusty byte[], bo exiftool nie miał co wyciągać z takiego zdjęcia. Trzeba zapisywać to zdjęcie nie klasą Image, a jako surowy ciąg bajtów.
2. Przez przypadek odkryłem też, że jeśli w ścieżce do exiftool.exe jest spacja to wszystko się posypie jeszcze bardziej. Odpalając command prompt z argumentem "/C" trzeba wszystko za tym /C zawrzeć w cudzysłowach, oraz każdą indywidualną ścieżke. Przykład w klasie ImageExtracter.

No i przy okazji zapomniałem 1 linię usunąć w serwerze... To już tu dorzuce.

MaxKolac committed:
Reverted DB changes in 3ccba2bd87b3636bd2806a230f16258441b649da

MaxKolac commented:
https://github.com/MaxKolac/zpi/pull/26/commits/1e5943da90badb1a03ee4e1f6e8086a1a614ecf4 
Dwie sprawy:
1. Okazuje się że klasa Image ma kompletnie w poważaniu metadane przy zapisywaniu. Metoda `Image.Save` zapisywała plik, który był faktycznie zdjęciem... ale bez metadanych. Jego rozmiar to był 28 KB, podczas gdy zdjęcia w folderze ExampleImages na ZPICameraSimulator mają po 2 MB. Dlatego Result tego jednego Task'a to był pusty byte[], bo exiftool nie miał co wyciągać z takiego zdjęcia. Trzeba zapisywać to zdjęcie nie klasą Image, a jako surowy ciąg bajtów.
2. Przez przypadek odkryłem też, że jeśli w ścieżce do exiftool.exe jest spacja to wszystko się posypie jeszcze bardziej. Odpalając command prompt z argumentem "/C" trzeba wszystko za tym /C zawrzeć w cudzysłowach, oraz każdą indywidualną ścieżke. Przykład w klasie ImageExtracter.
